File: src/generators/templates/ruby/wrapper.rb
# === Sandbox Executor Ruby Inline Wrapper ===
require 'json'

START_MARKER = "__SANDBOX_RESULT_START__"
END_MARKER = "__SANDBOX_RESULT_END__"

begin
  require_relative 'user_code'
  
  # Inline arguments
  args = {{ARGS}}
  kwargs = {{KWARGS}}
  func_name = "{{FUNCTION_NAME}}"
  
  if kwargs && !kwargs.empty?
    symbol_kwargs = kwargs.transform_keys(&:to_sym)
    result = send(func_name, *args, **symbol_kwargs)
  else
    result = send(func_name, *args)
  end

  output = {
    "success" => true,
    "result" => result
  }
rescue => e
  output = {
    "success" => false,
    "error" => {
      "message" => e.message,
      "type" => e.class.name,
      "stack" => e.backtrace.join("
")
    }
  }
end

puts START_MARKER
puts JSON.generate(output)
puts END_MARKER

File: src/generators/templates/php-cgi/wrapper.php
<?php
// === Sandbox Executor PHP Inline Wrapper ===

const START_MARKER = "__SANDBOX_RESULT_START__";
const END_MARKER = "__SANDBOX_RESULT_END__";

function __sandbox_serialize($obj) {
    if (is_object($obj)) {
        return (array)$obj;
    }
    return $obj;
}

try {
    @require_once '/workspace/user_code.php';
    
    // Inline arguments
    $args = {{ARGS}};
    $kwargs = {{KWARGS}};
    $funcName = "{{FUNCTION_NAME}}";
    
    if (!function_exists($funcName)) {
        throw new Exception("Function '$funcName' not found.");
    }

    if (!empty($kwargs)) {
        $result = call_user_func_array($funcName, array_merge($args, [$kwargs]));
    } else {
        $result = call_user_func_array($funcName, $args);
    }

    $output = [
        "success" => true,
        "result" => __sandbox_serialize($result)
    ];
} catch (Throwable $e) {
    $output = [
        "success" => false,
        "error" => [
            "message" => $e->getMessage(),
            "type" => get_class($e),
            "stack" => $e->getTraceAsString()
        ]
    ];
}

echo START_MARKER . "
";
echo json_encode($output) . "
";
echo END_MARKER . "
";
